### PR TITLE
WICKET-6531 Crash in Unsafe.getObject when running on the J9 VM

### PR DESCRIPTION
**Pull request by dawidss7**

### CODE REVIEW

File: wicket-core/src/main/java/org/apache/wicket/core/util/objects/checker/CheckingObjectOutputStream.java [hunk 1]
```diff
@@ -235,6 +235,9 @@ public String toString()
 	{
 		try
 		{
+
+			available = "Oracle Corporation".equals(System.getProperty("java.vm.vendor")); //WICKET-6531
```
Comment (martin-g) [inline wicket-core/src/main/java/org/apache/wicket/core/util/objects/checker/CheckingObjectOutputStream.java]: I am not sure this is a good solution.

```
$ jshell                                                                                                                                                                             qa  
|  Welcome to JShell -- Version 11.0.4
|  For an introduction type: /help intro

jshell> System.getProperty("java.vm.vendor")
$1 ==> "Ubuntu"

jshell> ⏎                             
```

I think you meant `java.specification.vendor` ?! 

```
System.getProperty("java.specification.vendor")
$2 ==> "Oracle Corporation"
```

Comment (martin-g) [reply wicket-core/src/main/java/org/apache/wicket/core/util/objects/checker/CheckingObjectOutputStream.java]: Please test this with J9

Comment (martin-g) [reply wicket-core/src/main/java/org/apache/wicket/core/util/objects/checker/CheckingObjectOutputStream.java]: @dawidss7 Ping!

Comment (dawidss7) [reply wicket-core/src/main/java/org/apache/wicket/core/util/objects/checker/CheckingObjectOutputStream.java]: I cheked this on several different VMs an operating systems and i don't think we can tell if there is an Oracle vm or not. 
`java.specification.vendor` is always `Oracle Corporation`
 
ORACLE JAVA 8 - UBUNTU
    java.specification.vendor = Oracle Corporation
    java.vendor = Oracle Corporation
    java.vm.name = Java HotSpot(TM) 64-Bit Server VM
    java.vm.specification.vendor = Oracle Corporation
    java.vm.vendor = Oracle Corporation

ORACLE JAVA 8 - WINDOWS
    java.specification.vendor = Oracle Corporation
    java.vendor = Oracle Corporation
    java.vm.name = Java HotSpot(TM) 64-Bit Server VM
    java.vm.specification.vendor = Oracle Corporation
    java.vm.vendor = Oracle Corporation

ORACLE JAVA 10 - WINDOWS
    java.specification.vendor = Oracle Corporation
    java.vendor = Oracle Corporation
    java.vm.name = Java HotSpot(TM) 64-Bit Server VM
    java.vm.specification.vendor = Oracle Corporation
    java.vm.vendor = "Oracle Corporation"

AMAZON CORRETTO 11 - WINDOWS
    java.specification.vendor = Oracle Corporation
    java.vendor = Amazon.com Inc.
    java.vm.name = OpenJDK 64-Bit Server VM
    java.vm.specification.vendor = Oracle Corporation
    java.vm.vendor = Amazon.com Inc.

IBM JDK 8 - AIX
    java.specification.vendor = Oracle Corporation
    java.vendor = IBM Corporation
    java.vm.name = IBM J9 VM
    java.vm.specification.vendor = Oracle Corporation
    java.vm.vendor = IBM Corporation

OPENJDK J9 8 - WINDOWS
    java.specification.vendor = Oracle Corporation
    java.vendor = Eclipse OpenJ9
    java.vm.name = Eclipse OpenJ9 VM
    java.vm.specification.vendor = Oracle Corporation
    java.vm.vendor = Eclipse OpenJ9

Comment (martin-g) [reply wicket-core/src/main/java/org/apache/wicket/core/util/objects/checker/CheckingObjectOutputStream.java]: In that case we cannot use this PR.
What exactly is the crash in J9 ?

Comment (dawidss7) [reply wicket-core/src/main/java/org/apache/wicket/core/util/objects/checker/CheckingObjectOutputStream.java]: The problem is exactly as described in [https://issues.apache.org/jira/browse/WICKET-6531](url)
Serializing BigDecimal causes J9 vm crash.

**Comment by martin-g** [comment]
Related to: https://github.com/apache/wicket/pull/385

**Comment by martin-g** [comment]
Fixed with #385

**Pull request closed without merging**